mypyc/build.py

```diff
@@ -266,40 +184,37 @@ def build_using_shared_lib(sources: List[BuildSource],
                            cfiles: List[str],
                            build_dir: str,
                            extra_compile_args: List[str],
-                           ) -> List[MypycifyExtension]:
+                           ) -> List[Extension]:
     """Produce the list of extension modules when a shared library is needed.
 
     This creates one shared library extension module that all of the
     others link against and then one shim extension module for each
     module in the build, that simply calls an initialization function
     in the shared library.
 
-    We treat the shared library as a python extension so that it is
-    cleanly processed by setuptools, but it isn't *really* a python C
-    extension module on its own.
+    The shared library is a python extension module that exports the
```

Comment:
Is "All of the others link against" above this the best way to explain this? The other modules import the shared library, right?  Maybe wordsmith the explanation bit, as it can be confusing for somebody who is familiar with the old implementation, at least.

Maybe mention somewhere that `lib_name` is the name of the shared library.